pkg/bundle/bundle.go

```diff
@@ -74,33 +68,71 @@ func NewProtobufBundle(pbundle *protobundle.Bundle) (*ProtobufBundle, error) {
 }
 
 func (b *ProtobufBundle) validate() error {
+	bundleVersion, err := getBundleVersion(b.Bundle.MediaType)
+	if err != nil {
+		return fmt.Errorf("error getting bundle version: %w", err)
+	}
+
+	// if bundle version is < 0.1, return error
+	if semver.Compare(bundleVersion, "v0.1") < 0 {
+		return fmt.Errorf("%w: bundle version %s is not supported", ErrUnsupportedMediaType, bundleVersion)
+	}
+
+	// fetch tlog entries, as next check needs to check them for inclusion proof/promise
 	entries, err := b.TlogEntries()
 	if err != nil {
 		return err
 	}
 
-	switch b.Bundle.MediaType {
-	case SigstoreBundleMediaType01:
+	// if bundle version == v0.1, require inclusion promise
+	if semver.Compare(bundleVersion, "v0.1") == 0 {
 		if len(entries) > 0 && !b.hasInclusionPromise {
 			return errors.New("inclusion promises missing in bundle (required for bundle v0.1)")
 		}
-	case SigstoreBundleMediaType02:
+	} else {
+		// if bundle version >= v0.2, require inclusion proof
 		if len(entries) > 0 && !b.hasInclusionProof {
 			return errors.New("inclusion proof missing in bundle (required for bundle v0.2)")
 		}
-	case SigstoreBundleMediaType03, SigstoreBundleMediaType03Legacy:
+	}
+
+	// if bundle version >= v0.3, require verification material to not be X.509 certificate chain (only single certificate is allowed)
+	if semver.Compare(bundleVersion, "v0.3") >= 0 {
 		certs := b.Bundle.VerificationMaterial.GetX509CertificateChain()
 
 		if certs != nil {
 			return errors.New("verification material cannot be X.509 certificate chain (for bundle v0.3)")
 		}
-	default:
-		return ErrIncorrectMediaType
+	}
+
+	// if bundle version is newer than v0.3, return error as this version is not supported
+	if semver.Compare(bundleVersion, "v0.3") > 0 {
```

Comment:
Do we need to handle v0.3.x? I saw we include a test for `v0.3.1` but I don't believe we'll include the patch version. I think this is fine as-is, but if we do want to handle patch versions, does this need to be >= v0.4?

Reply:
I turned on auto-merge so this didn't get addressed, but good point! I can file another quick PR.